### PR TITLE
Demisto lock bug

### PR DESCRIPTION
**Pull request by JudahSchwartz**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - We'll verify with customer

## Related Issues
fixes: [link to the issue](https://github.com/demisto/etc/issues/46349)

## Description
This adds the merge contexts functions to commonserver and adds synchronization on lock releases

## Minimum version of Cortex XSOAR
- [ ] 6.0.0
- [ ] 6.1.0
- [x] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by JudahSchwartz** [comment]
commit 61cc82c8841f4d288863660d9021488d0ca10cb2 - branch - demisto-lock-bug-dev-release was sent to the customer

File: Packs/Base/ReleaseNotes/1_18_18.md [hunk 1]
```diff
@@ -0,0 +1,4 @@
+
+#### Scripts
+##### CommonServer
+- Internal code improvements.
```
Comment (Itay4) [inline Packs/Base/ReleaseNotes/1_18_18.md:4]: lets add something here that it is related to integration context handling

File: Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml [hunk 2]
```diff
@@ -31,7 +31,6 @@ script:
       description: Additional information to provide for the lock instance
     - name: timeout
       description: Timeout (seconds) for wait on lock to be freed
-      defaultValue: "600"
```
Comment (Itay4) [inline Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml:34]: why did you remove that ?

File: Packs/DemistoLocking/ReleaseNotes/1_0_3.md [hunk 3]
```diff
@@ -0,0 +1,4 @@
+
+#### Integrations
+##### Demisto Lock
+- Fixed an issue where failing to release the lock was ignored. Now using retries.
```
Comment (Itay4) [inline Packs/DemistoLocking/ReleaseNotes/1_0_3.md:4]:
```suggestion
- Fixed an issue where failing to release the lock was ignored.
```

**Comment by Itay4** [comment]
@JudahSchwartz please note the lgtm alerts

File: Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml (hunk 2 above)
Comment (JudahSchwartz) [reply Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml:34]: It was overriding the timeout parameter

Comment (Itay4) [reply Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml:34]: not sure i follow, this is the timeout parameter

Comment (JudahSchwartz) [reply Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml:34]: we have a timeout config parameter set for the instance, and a timeout argument sent for the getlock command. It doesnt really make sense to have a default on the argument, since the default should be taken from the config parameter. Having a default on the argument overrode the config parameter

Comment (JudahSchwartz) [reply Packs/DemistoLocking/Integrations/DemistoLock/DemistoLock.yml:34]: meaning the timeout argument always had a value, so the config param was ignored

**Review by Itay4 — APPROVED** [review]
